### PR TITLE
Replace nondet values in the size of array types in the trace

### PR DESCRIPTION
**Pull request by romainbrenguier**
All nondet symbols should be replaced in the trace, but the case of the
size in the type of arrays was not handled.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: d662023d03432053d89f72f549f9813814801da7).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/105838527

File: src/goto-symex/build_goto_trace.cpp [hunk 1]
```diff
@@ -14,6 +14,7 @@ Author: Daniel Kroening
 #include "build_goto_trace.h"
 
 #include <util/arith_tools.h>
+#include <util/as_const.h>
```
Comment (tautschnig) [inline src/goto-symex/build_goto_trace.cpp]: That's not actually used, is it?

**Review by tautschnig — APPROVED** [review]

**Review by smowton — COMMENTED** [review]
Needs a test

**Comment by romainbrenguier** [comment]
@smowton 
> Needs a test

test added

**Review by smowton — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:warning:
This PR failed Diffblue compatibility checks (cbmc commit: 047cf247638ae84de3b13debe60cd70e0725bd4e).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/105896303
Status will be re-evaluated on next push.
Common spurious failures include: the cbmc commit has disappeared in the mean time (e.g. in a force-push); the author is not in the list of contributors (e.g. first-time contributors); compatibility was already broken by an earlier merge.

**Review by allredj — COMMENTED** [review]
:warning:
This PR failed Diffblue compatibility checks (cbmc commit: ca7cecfbd559a42f31bab37fbc547811c4dbd3eb).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/105973317
Status will be re-evaluated on next push.
Common spurious failures include: the cbmc commit has disappeared in the mean time (e.g. in a force-push); the author is not in the list of contributors (e.g. first-time contributors); compatibility was already broken by an earlier merge.

**Pull request merged**